apps/forms-flow-ai/forms-flow-web/src/components/FOI/FOIRequest/FOIRequest.js

```diff
@@ -278,416 +264,180 @@ const FOIRequest = React.memo(({userDetail}) => {
-  const tabclick = (evt, param) => {
-    let clickedOk = true;
-    if (editorChange && param !== 'Comments') {
-      if (window.confirm("Are you sure you want to leave? Your changes will be lost.")) {
-        clickedOk = true;
-        setEditorChange(false);
-        setRemoveComment(true);
-      }
-      else {
-        setEditorChange(true);
-        setRemoveComment(false);
-        clickedOk = false;
-        param = 'Comments';
-        document.getElementById(param).className += " active";
-        const elementsByName = document.getElementsByName(param);
-        var i;
-        for (i = 0; i < elementsByName.length; i++) {          
-            elementsByName[i].className += " active";        
-        }
+  React.useEffect(() => {
+    if (editorChange){
+      window.history.pushState(null, null, window.location.pathname);
+      window.addEventListener('popstate', handleOnHashChange);
+      window.addEventListener('beforeunload', handleBeforeUnload);
+      return () => {
+        window.removeEventListener('popstate', handleOnHashChange);
+        window.removeEventListener('beforeunload', handleBeforeUnload);
       }
     }
-    else {
+  }, [editorChange]);
+
+  const tabclick = (param) => {
+    if(param === 'Comments') {
       setRemoveComment(false);
+      changeTabLinkStatuses(param);
+      return;
     }
-    var i, tabcontent, tablinks;
+    
+    if(editorChange) {
+      confirmChangesLost(
+        () => {
+          setEditorChange(false);
+          setRemoveComment(true);
+          changeTabLinkStatuses(param);
+        },
+        () => {
+          setEditorChange(true);
+          setRemoveComment(false);          
+        }
+      )
+    } else {
 
-    tabcontent = document.getElementsByClassName("tabcontent");
-    for (i = 0; i < tabcontent.length; i++) {
-      tabcontent[i].style.display = "none";
-      tabcontent[i].className = tabcontent[i].className.replace(" active", "");
+      changeTabLinkStatuses(param);
+      
     }
 
-    tablinks = document.getElementsByClassName("tablinks");
-    for (i = 0; i < tablinks.length; i++) {
-      tablinks[i].className = tablinks[i].className.replace(" active", "");
-    }
-    document.getElementById(param).style.display = "block";
-    if (clickedOk)
-      evt.currentTarget.className += " active";
-    
   }
+
+  const changeTabLinkStatuses = (param) => {
+    setTabLinksStatuses({
+      ...initialStatuses,
+      [param]: {
+        ...tabLinksStatuses[param],
+        active: true,
+        display: true,
+      },
+    });
+  }
+
+
   const bottomTextArray = _requestStatus.split('|');
       
-  const userId = userDetail && userDetail.preferred_username
+  const userId = userDetail?.preferred_username
   const avatarUrl = "https://ui-avatars.com/api/name=Riya&background=random"
-  const name = `${userDetail && userDetail.family_name}, ${userDetail && userDetail.given_name}`
+  const fullName = `${userDetail?.family_name}, ${userDetail?.given_name}`
```

Comment:
This might evaluate as `false, false` which is still a string and would evaluate as true for something like if (`false, false`) which may still appear if used.